example/parquet_reader.cc

```diff
@@ -41,254 +39,227 @@ static string ByteArrayToString(const ByteArray& a) {
-
-    // j is the row, k is the column
-    int k = 0, j = 0;
-
-    // prints column name
-    for (j = 0; j < row_group.columns.size(); ++j) {
-      char *str = (char*)malloc(50);
-      assert(str);
-      strcpy(str, metadata.schema[j+1].name.c_str());
-      printf("%-" COL_WIDTH"s", str);
-      free(str);
+    for(vector<ColumnReader*>::iterator it = readers.begin(); it != readers.end(); it++) {
+      delete *it;
     }
+  }
+}
 
-    cout << "\n";
-
-
-    for (j = 0;j < row_group.columns.size(); ++j)
-      switch(type_array[j]) {
-        case Type::BOOLEAN:
-          printf("%-" COL_WIDTH"s","BOOLEAN");
-          break;
-        case Type::INT32:
-          printf("%-" COL_WIDTH"s","INT32");
-          break;
-        case Type::INT64:
-          printf("%-" COL_WIDTH"s","INT64");
-          break;
-        case Type::FLOAT:
-          printf("%-" COL_WIDTH"s","FLOAT");
-          break;
-        case Type::DOUBLE:
-          printf("%-" COL_WIDTH"s","DOUBLE");
-          break;
-        case Type::BYTE_ARRAY:
-          printf("%-" COL_WIDTH"s","BYTE_ARRAY");
-          break;
-        default:
-          continue;
-      }
-
-    cout << "\n";
+int main(int argc, char** argv) {
+  if (argc > 3) {
+    cerr << "Usage: parquet_reader [--only-stats] <file>" << endl;
+    return -1;
+  }
 
-    static string result;
-    char* str1;
+  string filename;
+  bool printContents = true;
 
-    for (k = 0; k < total_row_number; ++k) {
-      for (j = 0; j < row_group.columns.size(); ++j) {
-        switch(type_array[j]) {
-          case Type::BOOLEAN:
-            printf("%-" COL_WIDTH"d",((bool*)(((bool**)column_ptr)[j]))[k]);
-            break;
-          case Type::INT32:
-            printf("%-" COL_WIDTH"d",((int32_t *)(((int32_t **)column_ptr)[j]))[k]);
-            break;
-          case Type::INT64:
-            printf("%-" COL_WIDTH"ld",((int64_t *)(((int64_t **)column_ptr)[j]))[k]);
-            break;
-          case Type::FLOAT:
-            printf("%-" COL_WIDTH"f",((float*)(((float**)column_ptr)[j]))[k]);
-            break;
-          case Type::DOUBLE:
-            printf("%-" COL_WIDTH"lf",((double*)(((double**)column_ptr)[j]))[k]);
-            break;
-          case Type::BYTE_ARRAY:
-            result = ByteArrayToString( ((ByteArray*)(((ByteArray**)column_ptr)[j]))[k] );
-            str1 = (char*)malloc(result.size());
-            assert(str1);
-            strcpy(str1, result.c_str());
-            printf("%-" COL_WIDTH"s", str1);
-            free(str1);
-            break;
-          default:
-            continue;
-        }
-      }
-      cout << "\n";
-
-      // print ends
+  // Read command-line options
+  char *param, *value;
+  for (int i = 1; i < argc; i++) {
+    if ( (param = std::strstr(argv[i], "--only-stats")) ) {
+      printContents = false;
+    } else {
+      filename = argv[i];
     }
-
-    return column_ptr;
   }
 
-  fclose(file);
-  return NULL;
+  readParquet(filename, printContents);
```

Comment:
It would be great if we could turn this into a `ParquetFileReader::DebugPrint` on top of #20, then the parquet_reader.cc executable would only be a few lines of code.